test/auto_parallel/pir/pir_reshard_p_to_s.py

```diff
@@ -0,0 +1,300 @@
+
+        if self._shard == 1:
+            if self.rank != self._mesh.process_ids[-1]:
+                np.testing.assert_equal(main_program.num_ops(), 9)
+                std_ops = [
+                    "builtin.parameter",
+                    "dist_op.shard_tensor",
+                    "pd_op.transpose",
+                    "pd_op.full",
+                    "pd_op.full",
+                    "builtin.combine",
+                    "pd_op.concat",
+                    "pd_op.reduce_scatter",
+                    "pd_op.transpose",
+                ]
+                np.testing.assert_equal(
+                    ops,
+                    std_ops,
+                )
+            else:
+                np.testing.assert_equal(main_program.num_ops(), 13)
+                std_ops = [
+                    "builtin.parameter",
+                    "dist_op.shard_tensor",
+                    "pd_op.transpose",
+                    "pd_op.full",
+                    "pd_op.full",
+                    "builtin.combine",
+                    "pd_op.concat",
+                    "pd_op.reduce_scatter",
+                    'pd_op.full_int_array',
+                    'pd_op.full',
+                    'pd_op.split',
+                    'builtin.split',
+                    "pd_op.transpose",
+                ]
+                np.testing.assert_equal(
+                    ops,
+                    std_ops,
+                )
+
+        for op in main_program.global_block().ops:
+            if op.name() == "pd_op.reduce_scatter":
+                assert op.dist_attr.num_operands() == 1
+                assert op.dist_attr.num_results() == 1
+                assert op.dist_attr.process_mesh == self._mesh
+                op_operand_dist_attr = op.dist_attr.operand(
+                    0
+                ).as_tensor_dist_attr()
+                assert op_operand_dist_attr.process_mesh == self._mesh
+                assert op_operand_dist_attr.dims_mapping == [-1, -1]
+                assert op_operand_dist_attr.partial_status == {
+                    0: paddle.base.core.ReduceType.kRedSum
+                }
+                op_result_dist_attr = op.dist_attr.result(
+                    0
+                ).as_tensor_dist_attr()
+                assert op_result_dist_attr.process_mesh == self._mesh
+                if self._shard == 0:
+                    assert op_result_dist_attr.dims_mapping == [0, -1]
+                else:
+                    assert op_result_dist_attr.dims_mapping == [-1, 0]
+                assert op_result_dist_attr.partial_status == {}
+
+                op_value = op.result(0)
+                assert op_value.is_dense_tensor_type()
+                assert op_value.is_dist_dense_tensor_type()
+                assert op_value.dist_attr().process_mesh == self._mesh
+                if self._shard == 0:
+                    pass
+                else:
+                    assert op_result_dist_attr.dims_mapping == [-1, 0]
```

Comment:
we should check `op_value.dist_attr().dims_mapping == [-1, 0]`? not `op_result_dist_attr.dims_mapping`